specs/ceph_crud_flows.adoc

```diff
@@ -0,0 +1,557 @@
+// vim: tw=79
+
+= CRUD flows for ceph cluster entities
+
+The ceph-integration module need to provide flows for creating, retrieving,
+updating and deleting various entities like pools, rbds, ec-pools etc.
+
+== Problem description
+
+Currently the ceph-integration module provides flows only for creation and
+deletion of storage pools. Need to add flows for managing rbds, osd, erasure
+code profiles and ec-pools as well.
+
+== Use Cases
+
+CRUD operations for ceph cluster entities like storage pools, rbds, ec-pools and
+erasure code profiles.
+
+== Proposed change
+
+Add flows in ceph-integration module for below
+
+* Pool
+** Create
+** Update
+** Delete
+
+* RBDs
+** Create
+** Delete
+** Resize
+
+* Erasure coded pool
+** Create
+** Update
+** Delete
+
+* Erasure code profile
+** Create
+** Delete
+
+=== Alternatives
+
+There are two options to achieve the CRUD flows for ceph entities -
+
+* Use Calamari REST endpoints
+** pros
+*** Currently Calamari supports create/update/delete REST end points for pool
+management. For everything else, we need to invoke generic `CLI` end point from
+Calamari
+
+** cons
+*** Introduces another daemon on cluster MON nodes
+
+* Use direct rados and rnd command calls from ceph-integration modules
+** pros
+*** Doesn't need another daemon running on the MON nodes
+
+** cons
+*** None
+
+=== Data model impact:
+
+* Add models for
+** RBDs
+** Erasure coded pools
+** Erasure code profiles
+
+The snippet from definitions regarding the changes are as below
+
+```
+namespace.tendrl.ceph_integration:
+  flows:
+    .......
+    CreateECProfile:
+      atoms:
+        - tendrl.ceph_integration.objects.ECProfile.atoms.create.Create
+      help: "Create EC Profile"
+      enabled: true
+      inputs:
+        mandatory:
+          - ECProfile.name
+          - ECProfile.k
+          - ECProfile.m
+        optional:
+          - ECProfile.plugin
+          - ECProfile.directory
+          - ECProfile.ruleset_failure_domain
```

Comment:
The atoms are created for the time being but not exposed from tendrl UI. The default ec-profiles with 4+2, 6+3 and 8+4 would be created.